lua/jenkinsfile_linter.lua

```diff
@@ -6,20 +6,39 @@ local password = os.getenv("JENKINS_PASSWORD")
 local token = os.getenv("JENKINS_API_TOKEN") or os.getenv("JENKINS_TOKEN")
 local jenkins_url = os.getenv("JENKINS_URL") or os.getenv("JENKINS_HOST")
 local namespace_id = vim.api.nvim_create_namespace("jenkinsfile-linter")
-local insecure = os.getenv("JENKINS_INSECURE") and "--insecure" or ""
+local insecure = os.getenv("JENKINS_INSECURE") and "--insecure" or nil
 local validated_msg = "Jenkinsfile successfully validated."
 local unauthorized_msg = "ERROR 401 Unauthorized"
 local not_found_msg = "ERROR 404 Not Found"
 
+local function reject_nil(tbl)
+  return vim.tbl_filter(function(val)
+    return val ~= nil
+  end, tbl)
+end
+
+local function handle_error(msg)
+  if msg then
+    vim.notify("Something went wrong when trying to validate your file, check the logs.", vim.log.levels.ERROR)
+    log.error(msg)
+  end
+end
+
+local function handle_job_error(job)
```

Comment:
The job will report a summary of errors
https://github.com/nvim-lua/plenary.nvim/blob/2d9b06177a975543726ce5c73fca176cedbffe9d/lua/plenary/job.lua#L616-L622